### PR TITLE
Sentry Code Refactor

### PR DESCRIPTION
**Pull request by BKryth**
# About the pull request

Moves some of the code for the sentry laptop from #2100 to the item object.

# Explain why it's good for the game

Reusability, allows other item objects to have a special property of being slowly placed onto a table and having a variable dictate its new state. 

# Testing Photographs and Procedure

<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog

:cl:
refactor: moved the code for sentry laptop's table placement to a parent object so it can be used by other objects.
/:cl:

### CODE REVIEW

**Comment by BKryth** [comment]
Mainly did this so that I can implement the same table placement effect onto a future PR I'm making

File: code/modules/defenses/sentry_computer.dm [hunk 1]
```diff
@@ -99,36 +97,20 @@
...
-	else
-		to_chat(usr, SPAN_WARNING("You fail to setup the laptop"))
-
 /**
  * Called to reset the state of the laptop to closed and inactive.
  */
-/obj/item/device/sentry_computer/proc/teardown()
-	setup = FALSE
+/obj/item/device/sentry_computer/teardown()
+	..()
```
Comment (harryob) [inline code/modules/defenses/sentry_computer.dm:107]:
```suggestion
	. = ..()
```

**Review by harryob — CHANGES_REQUESTED** [review]
preserving the return type is a good habit (even if there isn't one)

File: code/game/objects/structures/tables_racks.dm [hunk 2]
```diff
@@ -305,9 +305,8 @@
 			to_chat(user, SPAN_WARNING("You slice at the table, but only claw it up a little."))
 		return
 
-	if (istype(W, /obj/item/device/sentry_computer))
-		var/obj/item/device/sentry_computer/computer = W
-		computer.setup(src)
+	if (W.has_special_table_placement)
+		W.set_to_table(src)
```
Comment (stanalbatross) [inline code/game/objects/structures/tables_racks.dm:309]: do this check before any istypes etc to save on proc overhead, easier/more generic checks should come first in big procs

**Review by stanalbatross — CHANGES_REQUESTED** [review]
just this one thing the rest is fine

File: code/game/objects/structures/tables_racks.dm (hunk 2 above)
Comment (BKryth) [reply code/game/objects/structures/tables_racks.dm]: done, moved to start, just below if(!W)

**Pull request merged**